RDKShell/RDKShell.cpp

```diff
@@ -6606,13 +6606,39 @@ namespace WPEFramework {
                         if (compositorName == previousFocusedClient)
                         {
                             std::cout << "setting the focus of " << compositorName << " to false " << std::endl;
-                            Exchange::IFocus *focusedCallsign = mCurrentService->QueryInterfaceByCallsign<Exchange::IFocus>(previousFocusedIterator->first);
-                            if (focusedCallsign != NULL)
+#ifdef HIBERNATE_SUPPORT_ENABLED
+                            //Check plugin state, skip for hibernated state
+                            string method = "status@" + previousFocusedIterator->first;
+                            Core::JSON::ArrayType<PluginHost::MetaData::Service> joResult;
+                            auto thunderController = getThunderControllerClient();
+                            uint32_t ret = thunderController->Get<Core::JSON::ArrayType<PluginHost::MetaData::Service>>(RDKSHELL_THUNDER_TIMEOUT, method.c_str(), joResult);
+                            bool isHibernated = false;
+                            if (ret == Core::ERROR_NONE && joResult.Length() > 0)
+                            {
+                                PluginHost::MetaData::Service service = joResult[0];
+                                if (service.JSONState == PluginHost::MetaData::Service::state::HIBERNATED)
```

Comment:
worth to consider excluding SUSPENDED state, only If we confirm the call to Focused(false) is not needed for hibernated or suspended apps